metaflow/plugins/events_decorator.py

```diff
@@ -0,0 +1,298 @@
+import json
+import time
+import re
+
+from metaflow import current
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+class TriggerDecorator(FlowDecorator):
+    """
+    Specifies the event(s) that this flow depends on.
+
+    ```
+    @trigger(event='foo')
+    ```
+    or
+    ```
+    @trigger(events=['foo', 'bar'])
+    ```
+
+    Additionally, you can specify the parameter mappings
+    to map event payload to Metaflow parameters for the flow.
+    ```
+    @trigger(event={'name':'foo', 'parameters':{'my_param': 'event_field'})
+    ```
+    or
+    ```
+    @trigger(events=[{'name':'foo', 'parameters':{'my_param_1': 'event_field_1'},
+                     {'name':'bar', 'parameters':{'my_param_2': 'event_field_2'}])
+    ```
+
+    Parameters
+    ----------
+    event : str or dict, optional
```

Comment:
To keep standard, please use Union[]

Reply:
this would be standard for typings, but I don't see Union[] being widely used in docstrings in the codebase so far. 
The "param: a or b" style seems much more prevalent at this point in the docstrings. Should we start steering away from it?